db/hints/manager.cc

```diff
@@ -1359,13 +1359,12 @@ class directory_initializer::impl {
             return make_ready_future<>();
         }
 
-        return with_semaphore(_lock, 1, [this] () {
+        return with_semaphore(_lock, 1, [this] () -> future<> {
```

Comment:
nit: s/corotinize/coroutinize/ in patch title (no need to respin the series just for that)